src/DataTypes/DataTypeEnum.cpp

```diff
@@ -262,6 +262,9 @@ void registerDataTypeEnum(DataTypeFactory & factory)
     factory.registerDataType("Enum8", createExact<DataTypeEnum<Int8>>);
     factory.registerDataType("Enum16", createExact<DataTypeEnum<Int16>>);
     factory.registerDataType("Enum", create);
+
+    /// MySQL
+    factory.registerAlias("ENUM", "Enum", DataTypeFactory::CaseInsensitive);
```

Comment:
This part does not need to be case insensitive. It's the actual enum values (the 'a', 'b', 'c' in `Enum('a','b','c')`) that need to be.

Reply:
I believe, that what was meant in https://github.com/ClickHouse/ClickHouse/issues/18310#issuecomment-752046182 
_(so ClickHouse does not have enum or ENUM data type, but it has Enum. We could add case-insensitive alias for Enum data type name (like we added BIGINT alias for Int64, for example)_
And without it we will have problem, because parser returns lowercase enum (and that's where the initial exception in issue is coming from)

Reply:
Right, then you can leave it of course :-)